tools/ci-scripts/linux/build_java.zsh

```diff
@@ -2,6 +2,11 @@
 
 set -eux
 
+# Java needs git tags to detect the version it's building. If we haven't
+# released in a while, the last tag might be many commits back, so we'll fetch
+# the entire history here.
+git fetch --unshallow --tags
```

Comment:
Do we need to `cd "$BOND_ROOT"` first? IIRC, we'll be in `$BUILD_ROOT` here. We'll see what happens with the build. :-)